### PR TITLE
[MERGE 9/20] feat(php-agent): Errors Inbox Improvement API Documentation

### PR DESCRIPTION
**Pull request by bduranleau-nr**
Addition of API documentation for the new API calls implemented as part of the Errors Inbox Improvement effort. 

The new APIs are:
- `newrelic_set_error_group_callback`, allowing a user to register a callback with the PHP Agent that will return a custom error group name when invoked by the agent during the handling of an error.
- `newrelic_set_user_id`, allowing an application to set a unique user ID string that associates an individual user with the current transaction.

Depends on https://github.com/newrelic/docs-website/pull/14640

### CODE REVIEW

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+## Return values
+
+API returns `true` if the callback is registered successfully, otherwise `false`.
+
+## Examples
+
+```php
+$callback = function(array $txndata, array $errdata) 
+{
+    // custom code to parse array data
```
Comment (ZNeumann) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:219]: Do we want to add a simple if/else to optionally set the fingerprint based on one of the data? This example would be more useful then, in my opinion.

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+## Return values
+
+API returns `true` if the callback is registered successfully, otherwise `false`.
+
+## Examples
+
+```php
+$callback = function(array $txndata, array $errdata) 
+{
```
Comment (ZNeumann) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:218]: Does this need the extention loaded check? https://github.com/newrelic/docs-website/pull/14460/files#diff-9ba4cabab0306f9cefe9fca9c30b048f091fb341b73f6a6300445d7920ef7fc2R70

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx (hunk 1 above)
Comment (bduranleau-nr) [reply src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx]: Made improvements in 4b8023d3a9fc8f754b2bb47dfa5f1086559bf3a7

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+---
+
+Group errors with your own custom-defined fingerprinting callback function.
+
+## Syntax
+
+```php
+newrelic_set_error_group_callback($callback)
+```
+
```
Comment (lavarou) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:21]: Documentation for other APIs have the syntax first and the description right below the syntax. For consistency's sake, I suggest to update docs for these two new APIs to match others.

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$txndata` (_array_)
```
Comment (lavarou) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:88]: PHP tends to use long, descriptive names for function arguments (see https://docs.newrelic.com/docs/apm/agents/php-agent/php-agent-api/newreliccustommetric-php-agent-api/):
```suggestion
        `$transaction_data` (_array_)
```

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+      </td>
+
+      <td>
+        Required. An array of transaction data provided to your callback by the PHP Agent.
+      </td>
+    </tr>
+    <tr>
+     <td>
+       `$errdata` (_array_)
```
Comment (lavarou) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:97]:
```suggestion
       `$error_data` (_array_)
```

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+     <td>
+       Required. An array of error data provided to your callback by the PHP Agent.
+     </td>
+    </tr>
+  </tbody>
+</table>
+
+### Array Key/Value pairs
+
+#### `$txndata` - PHP Agent transaction data provided to your callback
```
Comment (lavarou) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:108]:
```suggestion
#### `$transaction_data` - PHP Agent transaction data provided to your callback
```

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+        "status_code"
+      </td>
+      <td>
+        (_int_) HTTP status code (`200`, `404`, etc.)
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+#### `$errdata` - PHP Agent error data provided to your callback
```
Comment (lavarou) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:157]:
```suggestion
#### `$error_data` - PHP Agent error data provided to your callback
```

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx (hunk 3 above)
Comment (bduranleau-nr) [reply src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx]: Fixed in 2689de83d601ddb329e70f1170f4a2a04c26675a

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx (hunk 4 above)
Comment (bduranleau-nr) [reply src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx]: Applied changes in bb07f23bde149d3f160cd482fb45db3236b60519

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx (hunk 5 above)
Comment (bduranleau-nr) [reply src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx]: Applied changes in bb07f23bde149d3f160cd482fb45db3236b60519

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx (hunk 6 above)
Comment (bduranleau-nr) [reply src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx]: Applied changes in bb07f23bde149d3f160cd482fb45db3236b60519

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx (hunk 7 above)
Comment (bduranleau-nr) [reply src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx]: Applied changes in bb07f23bde149d3f160cd482fb45db3236b60519

**Review by lavarou — APPROVED** [review]
LGTM 👍

**Review by ZNeumann — APPROVED** [review]

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$callback` (_function_)
+      </td>
+
+      <td>
+        Required. Provide a callback function that will be registered with the PHP Agent.
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:66]:
```suggestion
        Required. Provide a callback function that will be registered with the PHP agent.
```

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+      <td>
+        Required. An array of transaction data provided to your callback by the PHP Agent.
+      </td>
+    </tr>
+    <tr>
+     <td>
+       `$error_data` (_array_)
+     </td>
+     <td>
+       Required. An array of error data provided to your callback by the PHP Agent.
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:100]:
```suggestion
       Required. An array of error data provided to your callback by the PHP agent.
```

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+     </td>
+    </tr>
+  </tbody>
+</table>
+
+### Array Key/Value pairs
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:106]:
```suggestion
### Array key/value pairs
```

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+
+#### `$transaction_data` - PHP Agent transaction data provided to your callback
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:108]:
```suggestion
#### `$transaction_data` - PHP agent transaction data provided to your callback
```

File: src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+        "status_code"
+      </td>
+      <td>
+        (_int_) HTTP status code (`200`, `404`, etc.)
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+#### `$error_data` - PHP Agent error data provided to your callback
```
Comment (ally-sassman) [inline src/content/docs/apm/agents/php-agent/php-agent-api/newrelic_set_error_group_callback.mdx:157]:
```suggestion
#### `$error_data` - PHP agent error data provided to your callback
```

**Review by ally-sassman — APPROVED** [review]

**Pull request merged**